server/plugin/virtual_agent.go

```diff
@@ -269,10 +328,33 @@ func (p *Plugin) CreateOutputLinkAttachment(body *OutputLink) *model.SlackAttach
 	}
 }
 
-func (p *Plugin) CreateOutputCardAttachment(body *OutputCardData) *model.SlackAttachment {
+func (p *Plugin) CreateOutputCardImageAttachment(body *OutputCardImageData) *model.SlackAttachment {
+	return &model.SlackAttachment{
+		Text:     fmt.Sprintf("**%s**\n%s", body.Title, body.Description),
+		ImageURL: body.Image,
+	}
+}
+
+func (p *Plugin) CreateOutputCardVideoAttachment(body *OutputCardVideoData) *model.SlackAttachment {
+	return &model.SlackAttachment{
+		Text: fmt.Sprintf("**[%s](%s)**\n%s", body.Title, body.Link, body.Description),
+	}
+}
+
+func (p *Plugin) CreateOutputCardRecordAttachment(body *OutputCardRecordData) *model.SlackAttachment {
+	var fields []*model.SlackAttachmentField
+	fields = append(fields, &model.SlackAttachmentField{
+		Title: body.Title,
+		Value: fmt.Sprintf("[%s](%s)", body.Subtitle, body.URL),
+	})
+	for _, field := range body.Fields {
+		fields = append(fields, &model.SlackAttachmentField{
```

Comment:
Now that you have a fixes size of the array, use indexes to put values instead of using `append`.